### PR TITLE
feat: add openapi express middleware to validate requests

### LINKED ISSUE #638: Validate Catalyst server with the API spec
We need to create a API spec module and include it on the Catalyst server so that we can validate that we are not breaking server API when introducing changes 

https://github.com/decentraland/catalyst-api-specs

### PR DESCRIPTION
**Pull request by guidota**
Resolves #638

### CODE REVIEW

File: comms/lighthouse/src/server.ts [hunk 1]
```diff
@@ -60,6 +62,13 @@ async function main() {
   if (accessLogs) {
     app.use(morgan('combined'))
   }
+  if (validateAPI) {
+    OpenApiValidator.middleware({
+      apiSpec: COMMS_API,
+      // validateResponses: true, // false by default
```
Comment (agusaldasoro) [inline comms/lighthouse/src/server.ts:68]: I'd never validate responses in code, only in tests

File: content/src/Environment.ts [hunk 2]
```diff
@@ -406,6 +407,8 @@ export class EnvironmentBuilder {
       () => new Map(Object.entries(process.env).filter(([name]) => name.startsWith('CACHE')))
     )
 
+    this.registerConfigIfNotAlreadySet(env, EnvironmentConfig.VALIDATE_API, () => process.env.VALIDATE_API !== 'false')
```
Comment (agusaldasoro) [inline content/src/Environment.ts:410]: Why do we need this to be configurable? Is it only for releasing it?

File: content/src/controller/Controller.ts [hunk 3]
```diff
@@ -412,7 +412,7 @@ export class Controller {
     // Path: /deployments
     // Query String: ?from={timestamp}&toLocalTimestamp={timestamp}&entityType={entityType}&entityId={entityId}&onlyCurrentlyPointed={boolean}&deployedBy={ethAddress}
 
-    const stringEntityTypes = this.asArray<string>(req.query.entityType as string)
+    const stringEntityTypes = this.asArray<string>(req.query.entityType as string | string[])
```
Comment (agusaldasoro) [inline content/src/controller/Controller.ts:415]: can this turn into an array of arrays?

File: content/test/integration/E2ETestEnvironment.ts [hunk 4]
```diff
@@ -65,7 +65,7 @@ export class E2ETestEnvironment {
   }
 
   async clearDatabases(): Promise<void> {
-    await this.repository.run((db) => db.query(`DROP SCHEMA ${E2ETestEnvironment.TEST_SCHEMA} CASCADE`), {
+    await this.repository.run((db) => db.query(`DROP SCHEMA IF EXISTS ${E2ETestEnvironment.TEST_SCHEMA} CASCADE`), {
```
Comment (agusaldasoro) [inline content/test/integration/E2ETestEnvironment.ts:68]: 👍

**Review by agusaldasoro — APPROVED** [review]

File: content/src/controller/Controller.ts (hunk 3 above)
Comment (guidota) [reply content/src/controller/Controller.ts:415]: As I can see in the method `asArray`, it shouldn't:

```ts
export function asArray<T>(elements: T[] | T): T[] {
  if (!elements) {
    return []
  }
  if (elements instanceof Array) {
    return elements
  }
  return [elements]
}
```

File: comms/lighthouse/src/server.ts (hunk 1 above)
Comment (menduz) [reply comms/lighthouse/src/server.ts:68]:
```suggestion
      validateResponses: process.env.CI == 'true' // or isRunningTests()
```

**Pull request merged**